magprime/automated_emails.py

```diff
@@ -15,7 +15,7 @@ def __init__(self, event):
     SeasonSupporterEmail(_event)
 
 
-AutomatedEmail(Attendee, 'MAGFest schedule, maps, and other FAQs', 'precon_faqs.html', lambda a: days_before(7, c.EPOCH))
+AutomatedEmail(Attendee, 'MAGFest schedule, maps, and other FAQs', 'precon_faqs.html', lambda a: days_before(7, c.EPOCH) and ((a.badge_status not in (c.INVALID_STATUS or c.DEFERRED_STATUS)) or (a.paid not c.NOT_PAID)))
```

Comment:
This is still the wrong syntax. the `not in` operator uses square brackets, not parentheses. It also uses comma delimiters instead of `or`. The correct syntax is `a.badge_status not in [c.INVALID_STATUS, c.DEFERRED_STATUS]`. I'm surprised this email even sent. You did test it, right?

Also, it may not have been clear from my comment, but if we check for "status is not deferred/invalid, OR they are not unpaid" then the email will still be sent to plenty of people who don't have a badge (particularly, staffers who are 'deferred' to next year have a paid status of `c.NEED_NOT_PAY`, so they'll get this email). It needs to be sent only to people who are both not deferred/invalid _and_ are not unpaid.